includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -241,6 +241,17 @@ public function get_product_data() {
 
 		$product = wc_get_product( $post->ID );
 
+		if ( 'variable' === $product->get_type() ) {
+			$attributes = wc_clean( wp_unslash( $_GET ) );
+
+			$data_store   = WC_Data_Store::load( 'product' );
+			$variation_id = $data_store->find_matching_product_variation( $product, $attributes );
+
+			if ( ! empty( $variation_id ) ) {
+				$product = wc_get_product( $variation_id );
+			}
+		}
```

Comment:
Ugh, I didn't realize we were **still** supporting WC 2.6. I remember talking about dropping it *so long ago* that I thought it already happened, [but](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/832) [no](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/833).

Thanks for challenging that assumption, we would have gotten complaints otherwise :)

I've made the required changes so it works for 2.6 in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1149/commits/f47d46b2840ed5025f96fcfdfb4b5daf263f18b6